### ISSUE TITLE
[Quilt 1.19.2] Game crashes and deletes config files when using number values

### ISSUE THREAD

**Issue by aninusmuffin** [EnnuiL/BoringDefaultGameRules#3]
Might be related to  #2
I want to use this mod in a modpack so I need to use "GENERATE_THIS"
[Crash log](https://mclo.gs/igGLeuv)

config.json :

```
{
	"$schema": "GENERATE_THIS",
	"default_game_rules": {
		"playersSleepingPercentage": 50
	},
	"generate_json_schema": true
}
```

**Comment by unilock** [comment]
Reading through the code led me to believe that removing the `$schema` and `generate_json_schema` values *entirely* might also work, leaving me with this:

```
{
	"default_game_rules": {
		"playersSleepingPercentage": 50,
		"creeperDestructionType": "DAMAGE_ONLY",
		"chargedCreeperDestructionType": "DAMAGE_ONLY",
		"canEndermenPlaceBlocks": false,
		"canEndermenPickBlocks": false
	}
}
```

I get the same crash + config file deletion as in the original post.

Note that I'm trying to use Boring Default Game Rules on a server. I think the issue may be due to the mod relying entirely on Mod Menu for configuration? Which is, of course, impossible in a server environment.

**Comment by EnnuiL** [comment]
oh my god, i'm so mad at myself for not looking at this issue before...
i apologize for the delay, but i can reproduce this issue, but what concerns me is the config file deletion (as described by #2, which is unfortunately too vague), because Quilt Config is also affected by it. I hope to investigate a fix for that problem and get a patch merged;
The issue here is that the JSON(5) deserializer is reading numbers as a BigDecimal because JSON accepts any valid number instead of attempting to fit it into an integer, double, etc. And yeah, the JSON(5) serializer ends not accounting for that at all! The mod has also failed to read those numbers previously, ~~and it's a miracle that I ended not finding any issues with that during testing.~~ Update: I ended figuring out that client testing that only involved the usage of the config screen meant that integers and doubles were used, and therefore, yeah... this was a result of a lack of server, and heck, client testing

Solution? I patched our own JSON serializer so it accepts BigIntegers and BigDecimals, and also I fixed the default game rule value editing in order to convert those numbers into integers or doubles. This fix will be out on a port of BDGR to 1.19.4 tomorrow (I plan to backport it to 1.19.2 too, it should be trivial);
Also, I plan to contribute the relevant fixes to Quilt Config, so these issues won't happen with anyone else choosing to use the JSON5 serializer, and also will prevent config data loss.

Finally? ~~It's `GENERATE_ME`, not `GENERATE_THIS`! :P~~ Update 2: augh, documentation is wrong! That will be fixed
once again, i'd like to apologize for the delay on answering this

**Comment by EnnuiL** [comment]
This has been fixed on Boring Default Game Rules 1.0.4 for Minecraft 1.19.4 with a backport to 1.19.2; I don't like doing backports often, but considering there was a serious issue overlooked for so long and also due to the fact it is a genuinely helpful mod for modpacks? I needed to do it, it's only fair to y'all
